### PR TITLE
adding support to certific features

### PR DESCRIPTION
**Pull request by diegomillan**
@felipemontoya @jfavellar90 

First commit allows to get a user info either by username or email

### CODE REVIEW

File: edunext_openedx_extensions/manage_api/views.py [hunk 1]
```diff
@@ -126,6 +129,51 @@ def post(self, request):
 
         return JsonResponse({"success": True}, status=201)
 
+    def get(self, request, **kwargs):  # pylint: disable=unused-argument
+        """
+        TODO: add me
```
Comment (felipemontoya) [inline edunext_openedx_extensions/manage_api/views.py]: Dont let this be forgoten

File: edunext_openedx_extensions/manage_api/views.py [hunk 2]
```diff
@@ -126,6 +129,51 @@ def post(self, request):
...
+        result, status = self._get_user(username, email)
+
+        return JsonResponse(result, status=status)
+
+    def _get_user(self, username, email):
+        """
+        Helper method to return the username and email of an existing user.
+        """
+        try:
+            existing_user = User.objects.get(username=username)
```
Comment (felipemontoya) [inline edunext_openedx_extensions/manage_api/views.py]: if you are using a method to get the user_by_email, out of consistency, you should use a method to get user_by_username, and you can test those separately

Comment (felipemontoya) [inline edunext_openedx_extensions/manage_api/views.py]: Do we currently use this API for the enrollment-api? I believe the answer is no. If this is the case, then my follow up question then would be: Are you planning to change the enroll_api to use only this API, or use the user-api at edxapp for the username based requests, and this for the email based requests?

**Comment by felipemontoya** [comment]
Also, please add tests for this

File: edunext_openedx_extensions/manage_api/views.py (hunk 2 above)
Comment (diegomillan) [inline edunext_openedx_extensions/manage_api/views.py]: Yes, the answer is no. I'm planning to use this API endpoint to decide if the user exists or not. I do not fire any requests to user-api at edxapp.

File: edunext_openedx_extensions/manage_api/views.py [hunk 3]
```diff
@@ -131,7 +131,7 @@ def post(self, request):
 
     def get(self, request, **kwargs):  # pylint: disable=unused-argument
         """
-        TODO: add me
+        Returns a registered edx-platform user searchin by username or email
```
Comment (felipemontoya) [inline edunext_openedx_extensions/manage_api/views.py]: typo: searching

File: edunext_openedx_extensions/manage_api/views.py [hunk 4]
```diff
@@ -143,7 +143,17 @@ def get(self, request, **kwargs):  # pylint: disable=unused-argument
 
     def _get_user(self, username, email):
         """
-        Helper method to return the username and email of an existing user.
```
Comment (felipemontoya) [inline edunext_openedx_extensions/manage_api/views.py]: this refactor makes no sense. You split the method into 2 but you are not separating any of the logic.The idea of my inital comment was to extract the lines inside of the try block.

**Comment by diegomillan** [comment]
@felipemontoya circle ci is passing, but the other checks not

**Comment by felipemontoya** [comment]
@diegomillan and do you think you can make your code better to make one or both of this checks pass?

Do you agree with the feedback given by the tool? is the cyclomatic complexity to high for the `get` method? is the code replication for _get_user_by_email and _get_user_by_username worth unifying?

**Comment by diegomillan** [comment]
@felipemontoya Yes, I think I can make an enhancement! Will try ASAP

**Comment by diegomillan** [comment]
@felipemontoya I fixed all quality problems

**Comment by diegomillan** [comment]
@jfavellar90

**Pull request merged**